### ISSUE TITLE
False negative `IgnoredReturnValue` with scope functions

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#6445]
## Steps to Reproduce
```kotlin
fun test(db: Database, sql: String) {
    db.insert(sql) // Detected

    with(db) {
        insert(sql) // Should be detected
    }

    db.let {
        it.insert(sql) // Should be detected
    }

    db.run {
        insert(sql) // Should be detected
    }
}

annotation class CheckResult

class Database {
    @CheckResult
    fun insert(query: String): Insert = TODO()
}

class Insert {
    fun execute(): Int = TODO()
}
```

## Your Environment
* Version of detekt used: 1.23.1
